File: KickbaseCore/Sources/KickbaseCore/Utilities/ScalingUtility.swift
//
//  ScalingUtility.swift
//  Kickbasehelper
//
//  Created for macOS scaling improvements
//

import SwiftUI

extension View {
    /// Appliziert plattformspezifische Skalierung
    public func macOSScaled() -> some View {
        self.modifier(MacOSScalingModifier())
    }
}

public struct MacOSScalingModifier: ViewModifier {
    @Environment(\.horizontalSizeClass) var horizontalSizeClass

    public init() {}

    var scaleFactor: CGFloat {
        #if os(macOS)
            return 1.2  // 20% größer auf macOS
        #elseif os(iOS)
            if UIDevice.current.userInterfaceIdiom == .pad {
                return 1.0  // iPad bleibt normal
            } else {
                return 1.0  // iPhone bleibt normal
            }
        #else
            return 1.0
        #endif
    }

    var fontScaleFactor: CGFloat {
        #if os(macOS)
            return 1.15  // 15% größere Schrift auf macOS
        #else
            return 1.0
        #endif
    }

    public func body(content: Content) -> some View {
        #if canImport(UIKit)
            let fontSize = UIFont.systemFontSize
        #elseif canImport(AppKit)
            let fontSize = NSFont.systemFontSize
        #else
            let fontSize = 14.0
        #endif

        return
            content
            .scaleEffect(scaleFactor)
            .font(.system(size: fontSize * fontScaleFactor))
    }
}

// Plattformspezifische Padding-Werte
extension View {
    public func adaptivePadding(_ edges: Edge.Set = .all, _ length: CGFloat? = nil) -> some View {
        let paddingValue = length ?? 8.0

        #if os(macOS)
            let macOSPadding = paddingValue * 1.2
            return self.padding(edges, macOSPadding)
        #else
            return self.padding(edges, paddingValue)
        #endif
    }

    public func adaptiveFont(_ style: Font.TextStyle) -> some View {
        #if os(macOS)
            return self.font(.system(style).weight(.regular))
        #else
            return self.font(.system(style))
        #endif
    }
}

// Zusätzliche macOS-spezifische Modifikationen
extension View {
    public func macOSOptimized() -> some View {
        self.modifier(MacOSOptimizationModifier())
    }
}

public struct MacOSOptimizationModifier: ViewModifier {
    public init() {}

    public func body(content: Content) -> some View {
        #if os(macOS)
            content
                .buttonStyle(PlainButtonStyle())  // Entfernt iOS-spezifische Button-Styles
        #else
            content
        #endif
    }
}

File: snapshots/skipstone/kickbasecore/KickbaseCore/destination/skipstone/KickbaseCore/src/main/kotlin/kickbase/core/ScalingUtility.kt
//
//  ScalingUtility.swift
//  Kickbasehelper
//
//  Created for macOS scaling improvements
//

package kickbase.core

import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.saveable.Saver
import androidx.compose.runtime.saveable.rememberSaveable
import androidx.compose.runtime.setValue
import skip.lib.*
import skip.lib.Set

import skip.ui.*
import skip.foundation.*
import skip.model.*

/// Appliziert plattformspezifische Skalierung
fun View.macOSScaled(): View = this.modifier(MacOSScalingModifier())

class MacOSScalingModifier: ViewModifier {
    internal var horizontalSizeClass: UserInterfaceSizeClass? = null

    constructor() {
    }

    internal val scaleFactor: Double
        get() = 1.0

    internal val fontScaleFactor: Double
        get() = 1.0

    override fun body(content: View): View {
        return ComposeBuilder l@{ composectx: ComposeContext ->
            val fontSize = 14.0

            return@l content
                .scaleEffect(scaleFactor)
                .font(Font.system(size = fontSize * fontScaleFactor)).Compose(composectx)
            ComposeResult.ok
        }
    }

    @Composable
    override fun Evaluate(content: View, context: ComposeContext, options: Int): kotlin.collections.List<Renderable> {
        this.horizontalSizeClass = EnvironmentValues.shared.horizontalSizeClass

        return super.Evaluate(content, context, options)
    }

    @androidx.annotation.Keep
    companion object {
    }
}

fun View.adaptivePadding(edges: Edge.Set = Edge.Set.all, length: Double? = null): View {
    val paddingValue = length ?: 8.0

    return this.padding(edges, paddingValue)
}

fun View.adaptiveFont(style: Font.TextStyle): View = this.font(Font.system(style))

fun View.macOSOptimized(): View = this.modifier(MacOSOptimizationModifier())

class MacOSOptimizationModifier: ViewModifier {
    constructor() {
    }

    override fun body(content: View): View {
        return ComposeBuilder { composectx: ComposeContext -> content.Compose(composectx) }
    }

    @androidx.annotation.Keep
    companion object {
    }
}
